test/StorageClient_test.dart

```diff
@@ -0,0 +1,521 @@
+      expect(searchResponse.result, []);
+      await storageClient.destroyDB();
+    });
+
+    //deleteNodes
+    test('Delete Infotainment node ', () async{
+      await prepareTree();
+      final deleteNodesResponse = await storageClient.deleteNodes(keyPartialInfotainment);
+      expect(deleteNodesResponse, isA<storage_api.StandardResponse>()); 
+      expect(deleteNodesResponse.success, isTrue); 
+
+      final searchResponse = await storageClient.search(keyPartialVehicle);
+      expect(searchResponse, isA<storage_api.ListResponse>()); 
+      expect(searchResponse.success, isTrue); 
+      expect(searchResponse.result, [
+        'Vehicle.Communication.Radio.Volume']);
+        await storageClient.destroyDB(); 
+    });
+
+    test('Delete Infotainment node (non-default namespace)', () async{
+      await prepareTree();
+      final deleteNodesResponse = await storageClient.deleteNodes(key4PartialInfotainmentTS);
+      expect(deleteNodesResponse, isA<storage_api.StandardResponse>()); 
+      expect(deleteNodesResponse.success, isTrue); 
+
+      final searchResponse = await storageClient.search(keyPartialVehicleTS);
+      expect(searchResponse, isA<storage_api.ListResponse>()); 
+      expect(searchResponse.success, isTrue); 
+      expect(searchResponse.result, [
+        'VehicleTS.Communication.Radio.Volume']);
+        await storageClient.destroyDB(); 
+    });
+
+    test('Delete Vehicle node ', () async{
+      await prepareTree();
+      final deleteNodesResponse = await storageClient.deleteNodes(keyPartialVehicle);
+      expect(deleteNodesResponse, isA<storage_api.StandardResponse>()); 
+      expect(deleteNodesResponse.success, isTrue); 
+
+      final searchResponse = await storageClient.search(keyPartialVehicle);
+      expect(searchResponse, isA<storage_api.ListResponse>()); 
+      expect(searchResponse.success, isTrue); 
+      expect(searchResponse.result, []);
+      await storageClient.destroyDB(); 
+    });
+
+    test('Delete node with empty key ', () async{
+      await prepareTree();
+      final deleteNodesResponse = await storageClient.deleteNodes(keyEmpty);
+      expect(deleteNodesResponse, isA<storage_api.StandardResponse>()); 
+      expect(deleteNodesResponse.success, isFalse); 
+
+      final searchResponse = await storageClient.search(keyPartialVehicle);
+      expect(searchResponse, isA<storage_api.ListResponse>()); 
+      expect(searchResponse.success, isTrue); 
+      expect(searchResponse.result, [
+        'Vehicle.Communication.Radio.Volume',
+        'Vehicle.Infotainment.HVAC.OutdoorTemperature',
+        'Vehicle.Infotainment.Radio.CurrentStation',
+        'Vehicle.Infotainment.Radio.Volume'
+          ]);
+      await storageClient.destroyDB(); 
+    });
+
+    test('listNodes Vehicle layer null ', () async{
+      await prepareTree();
+      late storage_api.SubtreeInfo  subtreeInfoVehicle1 = storage_api.SubtreeInfo(layers: null, node: 'Vehicle');
```

Comment:
why layers: null? layers should be optional, it should not be necessary to pass an explicit null here.